src/components/PopupWithForm.js

```diff
@@ -30,4 +32,22 @@ export default class PopupWithForm extends Popup {
     super.setEventListeners();
     this.formElement.addEventListener("submit", this._handleSubmitClick);
   }
+
+  setDeleteSubmitListener(data) {
+    super.setEventListeners();
+    this.formElement.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this.formSubmitHandler(data);
+      this.close();
+      this.formElement.reset();
+    });
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`